types/underscore/underscore-tests.ts

```diff
@@ -492,6 +492,50 @@ _.chain([{ a: 1 }, { a: 2 }, { a: 3 }, { b: 4 }])
     .contains(3)
     .value();
 
+// $ExpectType string[]
+_.chain({
+    a: {
+        common: 'only',
+        onlyA: 3
+    },
+    b: {
+        common: 'strings',
+        onlyB: true
+    },
+    c: {
+        common: 'strings',
+        onlyC: []
+    }
+})
+    .mapObject('common')
+    .reject(s => s === 'not')
+    .uniq()
+    .value();
+
+// $ExpectType any[][]
+_.chain({ one: 1, two: 2 })
+    .pairs()
+    .unzip()
+    .value();
+
+// $ExpectType [string, string | number | number[]] | undefined
```

Comment:
Interestingly TS seems to be willing to infer types from properties of simple object types but not interfaces, which admittedly isn't my favorite in this case but isn't terrible either I suppose.

```TS
interface Interface {
    a: string;
    b: number;
}

type Type = { a: string, b: number };

declare const interfaceNever: TypeOfDictionary<Interface>; // never
declare const typeUnion: TypeOfDictionary<Type>; // string | number
```

Reply:
That's a rather unpleasant suprise.